its/src/test/java/com/sonar/it/scanner/msbuild/utils/AnalysisContext.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * SonarScanner for .NET
+ * Copyright (C) 2016-2025 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package com.sonar.it.scanner.msbuild.utils;
+
+import com.sonar.it.scanner.msbuild.sonarcloud.CloudConstants;
+import com.sonar.it.scanner.msbuild.sonarqube.ServerTests;
+import com.sonar.orchestrator.Orchestrator;
+import java.nio.file.Path;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AnalysisContext {
```

Comment:
Can we make the type an [AutoCloseable](https://docs.oracle.com/javase/8/docs/api/java/lang/AutoCloseable.html)?

This could be used to remove the temp directory after the UT is finished. That way, we could remove the [`@TempDir`  annotations](https://github.com/SonarSource/sonar-scanner-msbuild/blob/f20cf948d7b60eef384f068e56a1ec00d05fee26/its/src/test/java/com/sonar/it/scanner/msbuild/sonarqube/ScannerMSBuildTest.java#L57C3-L58) from the ITs and integrate the clean up here and the `Path temp` workaround would no longer be needed.

Reply:
I was planning to deal with that in SCAN4NET-298, and based on your comment, I've extracted it to a separate SCAN4NET-362 as it will be nicer to have it isolated.

My original idea was to use jUnit extension, because I'll need it for the test name too.

But yes, the idea is to get rid of @TempDir